src/main.rs

```diff
@@ -48,44 +36,80 @@ fn main() {
         .arg("up")
         .spawn()
         .expect("up command did nto work");
-    up_command
+    up_command // Brings up the new can network
         .wait()
         .expect("Fail while waiting for up command");
+}
 
-    let mut stream = UnixStream::connect(ipc_path).unwrap();
-    let (tx, rx) = channel();
+/**
+ * Reads the can socket and publishes the data to the given client.
+ */
+fn read_can(mut publisher: Box<dyn Client + Send>) {
     //open can socket channel at name can0
     const CAN_CHANNEL: &str = "can0";
-    let socket = CANSocket::open(&CAN_CHANNEL);
+    let socket = CANSocket::open(CAN_CHANNEL);
```

Comment:
Why the removal of the &? Have we tested this?

Reply:
None of this has been tested but its changed anyways in jacks pr